lambdas/service/app.py

```diff
@@ -655,204 +661,237 @@ def repository_search(entity_type: str, item_id: str):
         raise NotFoundError(msg=e)
 
 
-@app.route('/repository/files', methods=['HEAD', 'GET'], cors=True)
-@app.route('/repository/files/{file_id}', methods=['GET'], cors=True)
+generic_object_spec = schema.object(additional_properties=True)
+array_of_object_spec = schema.array(generic_object_spec)
+hit_spec = schema.object(
+    additional_properties=True,
+    protocols=array_of_object_spec,
+    entryId=str,
+    samples=array_of_object_spec,
+    specimens=array_of_object_spec,
+    cellLines=array_of_object_spec,
+    donorOrganisms=array_of_object_spec,
+    organoids=schema.array(str),
+    cellSuspensions=array_of_object_spec,
+)
+
+page_spec = schema.object(
+    hits=schema.array(hit_spec),
+    pagination=generic_object_spec,
+    termFacets=generic_object_spec
+)
+
+
+def repository_filters_spec(facets):
+    return params.query(
+        'filters',
+        schema.optional(JSON),
+        content=schema.object(**{
+            facet: schema.optional(
+                schema.object(**{
+                    relation: schema.optional(schema.array({}))
+                    for relation in ('is', 'contains', 'within', 'intersects')
+                })
+            ) for facet in facets
+        }),
+        description=format_description('''
+            Filters to be applied when calling ElasticSearch. Omit
+            parameter or use `{}` to skip filtering. The relationship
+            parameters are always nested within a singleton array, so
```

Comment:
Not true. At least the "is" arrays can have more than one element. If the range-based operators only accept one that would constitute a bug.

"The relationship parameters are always nested within" is awkwardly expressed. Not really sure what you mean there.